src/models/expenses.model.js

```diff
@@ -0,0 +1,107 @@
+const fs = require('fs');
+const path = require('path');
+const { Expense } = require('../helpers/newExpense');
+const { getAllUsers } = require('./user.model');
+
+async function getAllExpenses() {
+  const fullPath = path.join(__dirname, '..', 'db', 'expenses.json');
+
+  const expenses = await fs.readFileSync(fullPath);
+
+  return JSON.parse(expenses.toString());
+}
+
+async function postExpense({
+  userId,
+  spentAt,
+  title,
+  amount,
+  category,
+  note = '',
+}) {
+  const { users } = await getAllUsers();
+
+  const user = users.find((item) => item.id === userId);
+
+  if (!user) {
+    return;
+  }
+
+  const newExpense = new Expense(
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  );
+
+  const { expenses } = await getAllExpenses();
+
+  expenses.push(newExpense);
+
+  const fullPath = path.join(__dirname, '..', 'db', 'expenses.json');
+
+  fs.writeFileSync(fullPath, JSON.stringify({ expenses }));
+
+  return newExpense;
+}
+
+async function getExpenseById(id) {
+  const { expenses } = await getAllExpenses();
+
+  return expenses.find((expense) => expense.id === id);
+}
+
+async function deleteExpenseById(id) {
+  let { expenses } = await getAllExpenses();
+
+  const isExpenseExist = expenses.find((expense) => expense.id === id);
+
+  if (!isExpenseExist) {
+    return false;
+  }
+
+  expenses = expenses.filter((expense) => expense.id !== id);
+
+  const fullPath = path.join(__dirname, '..', 'db', 'expenses.json');
+
+  fs.writeFileSync(fullPath, JSON.stringify({ expenses }));
+
+  return true;
+}
+
+async function updateExpenseById(id, data) {
+  let { expenses } = await getAllExpenses();
+  let updatedExpense;
+
+  expenses = expenses.map((expense) => {
+    if (expense.id === id) {
+      const newExpense = { ...expense, ...data };
+
+      updatedExpense = newExpense;
+
+      return newExpense;
+    }
+
+    return expense;
+  });
+
+  if (!updatedExpense) {
+    return;
+  }
+
+  const fullPath = path.join(__dirname, '..', 'db', 'expenses.json');
+
+  fs.writeFileSync(fullPath, JSON.stringify({ expenses }));
+
+  return updatedExpense;
+}
```

Comment:
In this case, you can try to use find and search currentExpense, then if an expense not found you return falsy value ( or nothing) without updating. If you find an expense map element update it and rewrite values.

Reply:
I just wanted to make it in one itteration)
so I will leave it as it is)

Reply:
if there is no expense I make this check) 
It would be undefined if no expense is found)

` if (!updatedExpense) {
    return;
  }`